examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+        verbose=0,
+    )
+
+    # Load best weights.
+    model.load_weights(file_path)
+
+    return model
+
+
+# Building model(s).
+instance_shape = train_data[0][0].shape
+models = [create_model(instance_shape) for _ in range(ENSEMBLE_AVG_COUNT)]
+
+# Show single model architecture.
+print(models[0].summary())
+
+# Training model(s).
+trained_models = [
+    train(train_data, train_labels, val_data, val_labels, model)
+    for model in tqdm(models)
+]
+
+"""
+## Model evaluation
+
+The models are in their deterministic state now and ready for evaluation.
+With each model we also create an associated intermediate model to get the
+weights from the attention layer.
+
+Based on the number of models (`ENSEMBLE_AVG_COUNT`), the models predict the results
+and then averaged out (equal contribution per model).
+"""
+
+
+def predict(data, labels, trained_models):
+
+    # Collect info per model.
+    models_predictions = []
+    models_attention_weights = []
+    models_losses = []
+    models_accuracies = []
+
+    for model in trained_models:
+
+        # Predict output classes on data.
+        predictions = model.predict(data)
+        models_predictions.append(predictions)
+
+        # Create intermediate model to get MIL attention layer weights.
+        intermediate_model = keras.Model(model.input, model.get_layer("alpha").output)
+
+        # Predict MIL attention layer weights.
+        intermediate_predictions = intermediate_model.predict(data)
+
+        # Reshape list of arrays.
+        attention_weights = np.squeeze(np.swapaxes(intermediate_predictions, 1, 0))
+        models_attention_weights.append(attention_weights)
+
+        loss, accuracy = model.evaluate(data, labels, verbose=0)
+        models_losses.append(loss)
+        models_accuracies.append(accuracy)
+
+    print(
+        f"The average loss and accuracy are {np.sum(models_losses, axis=0) / ENSEMBLE_AVG_COUNT:.2f}"
+        f" and {100 * np.sum(models_accuracies, axis=0) / ENSEMBLE_AVG_COUNT:.2f} % resp."
+    )
+
+    return (
+        np.sum(models_predictions, axis=0) / ENSEMBLE_AVG_COUNT,
+        np.sum(models_attention_weights, axis=0) / ENSEMBLE_AVG_COUNT,
+    )
+
+
+# Evaluate and predict classes and attention scores on validation data.
+class_predictions, attention_params = predict(val_data, val_labels, trained_models)
+
+# Plot some results of validation data bags per class.
+plot(
+    val_data,
+    val_labels,
+    "positive",
+    predictions=class_predictions,
+    attention_weights=attention_params,
+)
+plot(
+    val_data,
+    val_labels,
+    "negative",
+    predictions=class_predictions,
+    attention_weights=attention_params,
+)
+
+"""
+## Conclusion
+
+From the above plot, you can notice that the weights are always summing to 1. If it is a
+negative predicted bag, the weights will somehow be equally distributed. However, in a
+positive predict bag, the instance which resulted to the positve labelling, will have
```

Comment:
Lots of typos in this sentence. Suggest:

However, in a positively predict bag, the instance which resulted in the positive labelling will have a substantially higher attention score than the rest of the bag.

Reply:
Done.